### ISSUE TITLE
[HOLD for payment 2023-03-31] [$1000] https://new.expensify.com/concierge doesn't redirect to Concierge chat after sign-up/sign-in

### ISSUE THREAD

**Issue by marcochavezf** [Expensify/App#15785]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
Coming from this [thread](https://expensify.slack.com/archives/CDR85PSHM/p1678384183241449?thread_ts=1678310609.001019&cid=CDR85PSHM).

## Action Performed:
1. Ensure you're not logged in
2. Open https://new.expensify.com/concierge
3. Sign-up or sign-in

## Expected Result:
The user should be redirected to the Concierge chat

## Actual Result:
The user is redirected to the last opened chat

## Workaround:
Open Concierge manually or open https://new.expensify.com/concierge if the user is logged in

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** v1.2.80-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/6829422/224113686-7a88f0d4-1a32-4369-bc0b-1f08474b56b1.mov

**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01693e7548c561524b</li>
        <li>Upwork Job ID: 1634326085596753920</li>
        <li>Last Price Increase: 2023-03-10</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by mallenexpensify** [comment]
Checking to see if this should be put on hold pending Nav reboot
https://github.com/Expensify/App/issues/11768#issuecomment-1464236130

**Comment by marcaaron** [comment]
From a quick glance -  I'd guess it has more to do with some API issue than a navigation issue. Though, we need to understand the root cause in order to say for sure whether it is or isn't related.

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by MelvinBot** [comment]
Current assignee @marcochavezf is eligible for the Engineering assigner, not assigning anyone new.

**Comment by mallenexpensify** [comment]
Thanks @marcaaron , @marcochavezf , I added the engineering label, as you're already assigned, it's back to you.  Not sure if it should be internal or external

**Comment by mountiny** [comment]
I agree with Marc, I think this is not strictly an issue of the navigation, I will give this on the Callstack list for investigation

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01693e7548c561524b

**Comment by MelvinBot** [comment]
Current assignee @mallenexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mollfpr (`External`)

**Comment by MelvinBot** [comment]
Current assignee @marcochavezf is eligible for the External assigner, not assigning anyone new.

**Comment by mountiny** [comment]
@marcochavezf btw asked [here](https://expensify.slack.com/archives/C03UK30EA1Z/p1678488690050059) and they might get back with someone to look at this on Monday, thanks

**Comment by tienifr** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The user is redirected to the last opened chat rather than the Concierge chat when go to `/concierge` route when unauthenticated, then login.

### What is the root cause of that problem?
The root cause is that we're having the `ConciergePage` in the `PublicScreens` here https://github.com/Expensify/App/blob/2f72ad018ba55d29b106e42959f16955eef8fd70/src/libs/Navigation/AppNavigator/PublicScreens.js#L36. I believe the intention was to have the logic of `re-route to the login page` if the user is unauthenticated inside the `ConciergePage`, as mentioned in this comment https://github.com/Expensify/App/blob/2f72ad018ba55d29b106e42959f16955eef8fd70/src/pages/ConciergePage.js#L25.

When we go to the `/concierge` route, it's a public route so we don't have the logic of `saving the route and navigate to it after logging in`, thus this issue occurs.

### What changes do you think we should make in order to solve the problem?

We shouldn't have the  `ConciergePage` in the `PublicScreens` since we always need to be logged in to chat to Concierge, so it should be in the `AuthScreens` only. 

We need to remove the `ConciergePage` from the `PublicScreens` here https://github.com/Expensify/App/blob/2f72ad018ba55d29b106e42959f16955eef8fd70/src/libs/Navigation/AppNavigator/PublicScreens.js#L36, and optionally remove the `redirect to login` logic inside `ConciergePage` as well since it's not necessary now because that logic is built in when we navigate to any screen in the `AuthScreens` .

After this change, it will redirect to `/concierge` correctly after logging in.
 
### What alternative solutions did you explore?
NA

### Result
Working well after the fix:

https://user-images.githubusercontent.com/113963320/224472887-edc0d11f-23fd-44bd-a5d9-b7afbb3a345a.mov

**Comment by marcochavezf** [comment]
Ah @mountiny @mallenexpensify I already created a PR to fix this https://github.com/Expensify/App/pull/15788, the problem is that we're not handling the concierge route after the app is launched

**Comment by tienifr** [comment]
@marcochavezf what do you think about the solution [mentioned above](https://github.com/Expensify/App/issues/15785#issuecomment-1464856926)? I think the root cause there is more accurate and the fix is more straight-forward as well. 
> we're not handling the concierge route after the app is launched

This should be handled by default for all routes (not just Concierge), the issue is just that the Concierge screen is on the incorrect navigation stack (Public). So I don't think we should add custom handling logic for Concierge, just correcting its location in the navigation stack is enough.

**Comment by marcochavezf** [comment]
Hi @tienifr, thanks for stating the solution, but I think the `ConciergePage` was added to handle the deep link on native [platforms](https://github.com/Expensify/App/blob/704d9b3af43b3fb740649ddc426286a40c1259ba/src/libs/Navigation/linkingConfig.js#L31). Also for mobile web, we want to open the Concierge chat by default when the user opens the `/concierge` link. Could you also test the proposed solution and post screenshots for deep link `/concierge` working on native apps and mobile web?

**Comment by tienifr** [comment]
I have tested my proposal and here are screenshots for deep link `/concierge` working on native apps, mobile web and desktop also.

<details>
<summary>Mobile Web - Chrome</summary>


https://user-images.githubusercontent.com/113963320/224605533-18fb5781-d211-4fdb-b24f-6678848b15ac.mp4


</details>

<details>
<summary>Mobile Web - Safari</summary>


https://user-images.githubusercontent.com/113963320/224605555-3a3bc8db-65a7-4d9e-ba50-6b169b40df8e.mp4


</details>

<details>
<summary>Desktop</summary>


https://user-images.githubusercontent.com/113963320/224605566-b2572ea9-6428-49cb-8f35-3454163d0228.mp4



</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/113963320/224605582-77350759-c03a-44b1-9ace-d8e2ca9c9fa9.mp4

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/113963320/224605612-5c4cc332-5878-488d-9c8d-0196483858d4.mp4

</details>

**Comment by MelvinBot** [comment]
@marcochavezf, @mallenexpensify, @mollfpr Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by marcochavezf** [comment]
@tienifr! whoa, sorry for the delay here and thanks for testing it! Yeah, looks like that's the cleanest solution, let's do it! 🚀

**Comment by MelvinBot** [comment]
📣 @tienifr You have been assigned to this job by @marcochavezf!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by tienifr** [comment]
Hi @marcochavezf @mollfpr the [PR](https://github.com/Expensify/App/pull/15974) is ready for review!

**Comment by mallenexpensify** [comment]
@tienifr @mollfpr can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~01693e7548c561524b

**Comment by mollfpr** [comment]
@mallenexpensify accepted, thank you!

**Comment by tienifr** [comment]
Hi @marcochavezf @mollfpr, I created [the PR](https://github.com/Expensify/App/pull/15974) and I'm waiting for the review. Thank you!

**Comment by tienifr** [comment]
# Regression Test Proposal

**Bug**: Couldn’t copy by command C on the placeholder in the Search Page after selecting it

**Proposed Test Steps**:


1. Ensure you're not logged in
2. Open https://new.expensify.com/concierge
3. Sign-up or sign-in
4. Verify that the user should be redirected to the Concierge chat



Do we :+1: or :-1:

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.87-1 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15974
- https://github.com/Expensify/App/pull/15788


If no regressions arise, payment will be issued on 2023-03-28. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@mollfpr / @marcochavezf] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@mollfpr / @marcochavezf] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@mollfpr / @marcochavezf] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@mallenexpensify] Determine if we should create a regression test for the bug.
- [ ] [@mollfpr] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to the [appropriate location](https://github.com/Expensify/App/blob/main/contributingGuides/TestRail_Navigation_C%2B.md) to ensure the same bug will not reach production again.
- [ ] [@mallenexpensify] Review the proposed regression test steps and location.
- [ ] [@mallenexpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@mollfpr / @marcochavezf] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@mollfpr / @marcochavezf] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@mollfpr / @marcochavezf] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@mallenexpensify] Determine if we should create a regression test for this bug.
- [ ] [@mollfpr] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@mallenexpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by mallenexpensify** [comment]
This issue's a lil tricky but.. it appears 
@tienifr assigned on March 15th. 
Merged on March 16th. 

From what I can tell, the other linked issue and PR don't affect payment here, please comment if that's not accurate. So both @tienifr and @mollfpr should be due $1500 which includes the $500 timeliness bonus. 
 
Gonna leave this open for a couple days before I pay to be sure the above is accurate. Everyone's already hired for the job [here](https://www.upwork.com/jobs/~01693e7548c561524b)

**Comment by MelvinBot** [comment]
@marcochavezf, @mallenexpensify, @mollfpr, @tienifr Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by marcochavezf** [comment]
For this one, the proposed solution was merged, but there was an [uncovered edge case](https://github.com/Expensify/App/issues/16080) that was fixed in another PR. The steps to catch that case were a bit tricky, so I agree we can consider it as a different issue and don't affect payment here.
